### PR TITLE
new cask: starnet++

### PR DESCRIPTION
**Pull request by audibleblink**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by miccal** [comment]
Ping @Homebrew/cask and @Homebrew/maintainers for your opinions, please.

**Comment by MikeMcQuaid** [comment]
@miccal can you elaborate a bit more on what you're looking for opinions on? Do you have particular concerns here?

**Comment by miccal** [comment]
No concerns, just wanted more opinions.

**Comment by carlocab** [comment]
Packaging this as a cask seems fine, but the choice to install into `HOMEBREW_PREFIX/share` seems very odd. That decision seems to contribute also to at least some of the hacks in the shimscript.

File: Casks/starnet-plus-plus.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    end
+  end
+
+  share = "#{HOMEBREW_PREFIX}/share"
+  libs = [
+    "starnet++",
+    "libtensorflow.2.dylib",
+    "libtensorflow_framework.2.dylib",
+    "starnet#{version.csv.first}_weights.pb",
+  ]
```
Comment (carlocab) [inline Casks/starnet-plus-plus.rb:26]:
```suggestion
```

File: Casks/starnet-plus-plus.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      cleanup() {
+        rm -f starnet#{version.csv.first}_weights.pb
+      }
+      trap cleanup RETURN EXIT SIGINT SIGKILL
+
+      # the binary hardcodes the weights path so we have to symlink it to the CWD
+      ln -sf "#{share}/starnet#{version.csv.first}_weights.pb" .#{" "}
+
+      # define a load path since the libs are not in the same dir as the bin
+      DYLD_LIBRARY_PATH=#{share} command "#{share}/starnet++" "$@"
```
Comment (carlocab) [inline Casks/starnet-plus-plus.rb:45]:
```suggestion
      exec "#{staged_path}/starnet++" "$@"
```

File: Casks/starnet-plus-plus.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    EOS
+  end
+
+  postflight do
+    set_permissions "#{staged_path}/StarNetv#{version.csv.first}CLI_MacOS/starnet++", "0755"
+    libs.each do |af|
+      system_command "xattr", args: ["-c", "#{staged_path}/StarNetv#{version.csv.first}CLI_MacOS/#{af}"]
+      File.symlink "#{staged_path}/StarNetv#{version.csv.first}CLI_MacOS/#{af}", "#{share}/#{af}"
```
Comment (carlocab) [inline Casks/starnet-plus-plus.rb:53]:
```suggestion
```

File: Casks/starnet-plus-plus.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    end
+  end
+
+  uninstall_postflight { File.unlink(*libs.map { |af| "#{share}/#{af}" }) }
```
Comment (carlocab) [inline Casks/starnet-plus-plus.rb:57]:
```suggestion
```

**Review by carlocab — COMMENTED** [review]
I feel like something like this might work? Please test it first, though.

**Comment by p-linnane** [comment]
@audibleblink Please rebase instead of using merge commits.

**Comment by audibleblink** [comment]
@carlocab your suggestion did not work as intended. The main program binary hardcodes the location of the ML weights, so the symlink is required. I also did not want to taint the homebrew lib directory, since the program is looking for bundled tensorflow libs, and tensorflow is already a formula. This is the reason I have the libs and weights in a share directory, and setting the `DYLD_LIBRARY_PATH` variable in the shim script

**Comment by carlocab** [comment]
> The main program binary hardcodes the location of the ML weights, so the symlink is required. 

You've said this, but never explained where this location is hard-coded to.

> I also did not want to taint the homebrew lib directory, since the program is looking for bundled tensorflow libs, and tensorflow is already a formula.

I agree. That's why I did not suggest installing anything into the lib directory.

> This is the reason I have the libs and weights in a share directory, and setting the `DYLD_LIBRARY_PATH` variable in the shim script

`share` is for [architecture-independent data](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s11.html), which this is not. Why do we need to install into `share` at all?

**Comment by audibleblink** [comment]
This program looks for the weights in the process' $cwd.

You're right. First time committing and got a little too excited before completely understanding the  framework y'all have built. Hopefully this is more satisfactory.

**Comment by miccal** [comment]
Thank you @audibleblink.

**Review by miccal — APPROVED** [review]

**Pull request merged**